### DISCUSSION TITLE
`defineLoader` (for single fetch) and custom context

### DISCUSSION THREAD

**Discussion by DerJacques** [remix-run/remix#9508] [Q&A]
I'm currently investigating the new single fetch functionality and am running into type issues with my custom context provided via the express adapter.

Previously, I had something like this:

```
export const loader = async ({ context: { somethingAddedViaExpress } }: MyLoaderArgs) => {
```

In order to get the right types for single fetch loaders, I need to use the new `defineLoader` method. 

This does **not** work:

```
export const loader = defineLoader(async ({ context: { somethingAddedViaExpress } }: MyLoaderArgs) => {
```

Error:
```
Property 'somethingAddedViaExpress' is missing in type 'AppLoadContext' but required in type 'MyAppLoadContext'
```

In an ideal world, I would love for `defineLoader` to take a generic that allows me to provide the context type. Is there any other way I could achieve this?

**Comment by sergiodxa** [comment]
You need to customize the AppLoadContext in your app, then defineLoader should give you the correct type, see https://sergiodxa.com/tutorials/customize-remix-app-load-context-type
